lib/active_triples/rdf_source.rb

```diff
@@ -547,6 +547,19 @@ def from_uri(uri, vals = nil)
       end
 
       ##
+      # Apply a data model using a given strategy.
+      # 
+      # @param [ActiveTriples::DataModel, #properties] data_model A data model 
+      #   to apply.
+      # @param [#apply!] application_strategy A strategy for applying. Defaults
+      #   to ActiveTriples::ApplicationStrategy
+      def apply_data_model(data_model, application_strategy=ActiveTriples::ApplicationStrategy)
```

Comment:
In my mind this is a command, not a query, and doesn't necessitate a return value (I mean, there is one, but it's not be depended on). The failure case would be exceptions - the application strategy doesn't work for the class you're calling it on or the data model you pass it doesn't respond to #properties.